### PR TITLE
Allow change of templating language when reinstalling

### LINKED ISSUE #191: Changing `[jinja2:suite.rc]` to `[template variables]` causes reinstall to fail
**Describe the bug**

Consider a placeholder workflow with `rose-suite.conf` containing a `[jinja2:suite.rc]` section. Updating this to `[template variables]` and reinstalling raises the following:

```
PluginError: Error in plugin cylc.post_install.rose_opts
MultipleTemplatingEnginesError: You should not define more than one templating section. You defined:
	jinja2:suite.rc; template variables
```

**Steps to reproduce the bug**
```bash
cd $(mktemp -d)
touch flow.cylc
echo "[jinja2:suite.rc]" > rose-suite.conf
cylc install -n tmp
echo "[template variables]" > rose-suite.conf
cylc reinstall tmp
```

**Expected behavior**
This error should be raised only if the _resulting config_ has _both_ sections.

**Pull requests welcome!**
This is an Open Source project - please consider contributing a bug fix
yourself (please read `CONTRIBUTING.md` before starting any work though).

Issue raised on behalf of @GAndrsn

### PR DESCRIPTION
**Pull request by wxtim**
These changes close #191 

in `record_cylc_install_options` the function `identify_templating_section` is used to check whether the user has set more than one templating section. This is done after the previous `rose-suite-cylc-install.conf` has been merged with the new one, so that if the user has changed the templating language both will be present in the merged config and the test will always fail.

## Work done
- Move checks for multiple templating sections to before old and new configs are merged
- Algorithm merging previous and current config  renames old template variables section if new install has changes it.
- Update tests for merge algorithm.
- Remove test that it's impossible to change templating languages on reinstall - it should not be.

## Requirements check-list

- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Applied any dependency changes to `setup.cfg`.
- [x] Appropriate tests are included (unit and/or functional).
- [x] Appropriate change log entry included.
- [x] No documentation update required.

### CODE REVIEW

File: cylc/rose/utilities.py [hunk 1]
```diff
@@ -42,6 +42,9 @@
 ROSE_ORIG_HOST_INSTALLED_OVERRIDE_STRING = (
     ' ROSE_ORIG_HOST set by cylc install.'
 )
+TEMPLATING_SECTIONS = [
+    'jinja2:suite.rc', 'empy:suite.rc', 'template variables']
+JINJA2_SECTION, EMPY_SECTION, TV_SECTION = TEMPLATING_SECTIONS
```
[the remark this answers is not in the document]
Comment (wxtim) [reply cylc/rose/utilities.py:47]: Yes, half done refactor

```suggestion
```

**Review by MetRonnie — APPROVED** [review]

File: cylc/rose/entry_points.py [hunk 2]
```diff
@@ -221,7 +224,6 @@ def record_cylc_install_options(
     rose_suite_conf = add_cylc_install_to_rose_conf_node_opts(
         rose_suite_conf, cli_config
     )
-    identify_templating_section(rose_suite_conf)
```
Comment (oliver-sanders) [inline cylc/rose/entry_points.py:224]: Shouldn't we continue to check the merged config?

**Review by oliver-sanders — APPROVED** [review]
One question, feel free to resolve/merge if happy.

**Comment by oliver-sanders** [comment]
@wxtim

File: cylc/rose/entry_points.py (hunk 2 above)
Comment (wxtim) [reply cylc/rose/entry_points.py:224]: Yes. I've put it back.

**Comment by wxtim** [comment]
> @wxtim

Will do once the tests have re-run

**Pull request merged**